config/docker-entrypoint.sh

```diff
@@ -80,44 +94,37 @@ if [ -n "$TRIAL" ]; then
 
 fi
 
-
 # Process arguments
-while [ "$#" != "0" ] ; do
-    case "$1" in
-    create)
-        handle () {
-            # Note, this *must* be create for `generateName` to work properly
-            kubectl create -f -
-        }
-
-        shift
-        ;;
-    delete)
-        handle () {
-            kubectl delete -f -
-        }
-
-        waitFn () {
-            if [ -n "$TRIAL" ] && [ -n "$NAMESPACE" ] ; then
-                kubectl wait pods --for=delete --namespace "$NAMESPACE" --selector "stormforge.io/trial=$TRIAL,stormforge.io/trial-role=trialResource" --timeout=5m
-            fi
-        }
-
-        shift
-        ;;
-    build)
-        handle () { cat ; }
-        waitFn () { :; }
-        shift
-        ;;
-    *)
-        echo "unknown argument: $1"
-        exit 1
-        ;;
-    esac
-done
+case "${MODE}" in
+  create)
+    handle () {
+        # Note, this *must* be create for `generateName` to work properly
+        kubectl create -f -
+    }
+
+    ;;
+  delete)
+    handle () {
+        kubectl delete -f -
+    }
+
+    waitFn () {
+        if [ -n "$TRIAL" ] && [ -n "$NAMESPACE" ] ; then
+            kubectl wait pods --for=delete --namespace "$NAMESPACE" --selector "stormforge.io/trial=$TRIAL,stormforge.io/trial-role=trialResource" --timeout=5m
+        fi
+    }
+
+    ;;
+  build)
+    handle () { cat ; }
+    waitFn () { :; }
```

Comment:
This `waitFn` is redundant with the new declaration on line 4. I wonder if there should be a comment on the delete and Prom `waitFn` that the no-op is being redefined?

Another option would be to change the last line to `[[ $(type -t waitFn) == function ]] && waitFn` and skip the no-op declaration, not sure which is more clear.